lib/ssh.ts

```diff
@@ -0,0 +1,90 @@
+import fs from 'fs';
+import os from 'os';
+
+/**
+ * Sets up the SSH credentials for the pipeline.
+ * This function copies the SSH identity file and known hosts file from the Bitbucket Pipelines agent to the local .ssh directory.
+ * It also updates the SSH configuration file to include the identity file.
+ *
+ * @throws {Error} If the SSH identity file cannot be copied or the SSH configuration file cannot be updated.
+ */
+export async function setupSshCredentials(): Promise<void> {
+    const homeDir = os.homedir();
+    const sshDir = `${homeDir}/.ssh/`;
+    const sshConfigDir = `/opt/atlassian/pipelines/agent/ssh`;
+    const identityFile = `${sshConfigDir}/id_rsa_tmp`;
+    const knownHostsFile = `${sshConfigDir}/known_hosts`;
+
+    // Ensure the SSH directory exists
+    const sshDirExists = await fs.promises
+        .stat(sshDir)
+        .then((stat) => stat.isDirectory());
+    if (!sshDirExists) {
+        await fs.promises.mkdir(sshDir, { recursive: true });
+    }
+
+    // Copy over the SSH identity file that Bitbucket has generated, if this fails then we should fail the whole pipeline
+    try {
+        console.log('Attempting to copy SSH identity file...');
+
+        const pipelinesIdFile = `${sshDir}/pipelines_id`;
+        await fs.promises.copyFile(identityFile, pipelinesIdFile);
+
+        console.log(`Copied to ${pipelinesIdFile}`);
+        console.log(`Adding identity file config to config file`);
+
+        const configFile = `${sshDir}/config`;
+        await fs.promises.appendFile(
+            configFile,
+            `IdentityFile ${pipelinesIdFile}`
+        );
+    } catch (e) {
+        console.error(
+            'Failed to update SSH configuration, check that SSH key configuration in Pipelines is valid. \n Check Pipelines -> SSH Keys.'
+        );
+        throw e;
```

Comment:
Can we just gracefully continue here? We don't _need_ SSH keys to deploy, it's only needed if installing packages via ssh

Reply:
That's a good point :sweat_smile: